test/model/src/core/kv_table.cpp

```diff
@@ -195,15 +204,7 @@ kv_table::insert(
 int
 kv_table::remove(const data_value &key, timestamp_t timestamp)
 {
-    kv_table_item *item = item_if_exists(key);
-    if (item == nullptr)
-        return WT_NOTFOUND;
-    try {
-        item->add_update(std::move(kv_update(NONE, fix_timestamp(timestamp))), true, false);
-        return 0;
-    } catch (wiredtiger_exception &e) {
-        return e.error();
-    }
+    with_transaction(return remove(txn, key), timestamp);
```

Comment:
A point for discussion rather than a request to change anything - this is fine as-is.

It would be more C++-ish to have a function `with_transaction` that accepts a function pointer to a function taking a transaction as an argument. That removes the "where did the `txn` variable come from" question that a reader would have while going over this code, and puts the return in a more natural place. Usage would look like this:

```
return with_transaction([&](auto txn) {
    return remove(txn, key);
}, timestamp);
```

But IMO this is more verbose even if it's more canonical or whatever.

Reply:
I think this would work—I just updated the PR with this. Can you please review?